air-quality-ui/system_tests/pages/city_page.ts

```diff
@@ -37,22 +80,29 @@ export class VairifyCityPage {
   }
 
   async setupRioDeJaneiroRoute() {
-    const genQuery = '*/**/air-pollutant/forecast*'
-    await this.page.route(genQuery, async (route) => {
-      await route.fulfill({ json: this.apiForecastAqi })
+    await this.page.route('*/**/air-pollutant/forecast*', async (route) => {
+      await route.fulfill({ json: this.mockForecastObject })
+    })
+    await this.page.route('*/**/air-pollutant/measurements*', async (route) => {
+      await route.fulfill({ json: this.mockMeasurementsObject })
     })
   }
 
-  async waitForGraphVisible() {
+  async waitForAllGraphsToBeVisible() {
     await this.aqiChart.waitFor({ state: 'visible' })
+    await this.pm2_5Chart.waitFor({ state: 'visible' })
+    await this.pm10Chart.waitFor({ state: 'visible' })
+    await this.o3Chart.waitFor({ state: 'visible' })
+    await this.no2Chart.waitFor({ state: 'visible' })
   }
+
   async gotoRioCityPage() {
     await this.page.goto('/city/Rio%20de%20Janeiro')
```

Comment:
Is this used more than once? If not I'd just do the page.goto in your setup function